controllers/activemqartemis_jdbc_ha_test.go

```diff
@@ -0,0 +1,283 @@
+				peerA.Spec.ResourceTemplates = []brokerv1beta1.ResourceTemplate{
+					{
+						Selector: &brokerv1beta1.ResourceSelector{
+							Kind: &kindMatchSs,
+						},
+						Patch: &unstructured.Unstructured{Object: map[string]interface{}{
+							"kind": "StatefulSet",
+							"spec": map[string]interface{}{
+								"template": map[string]interface{}{
+									"spec": map[string]interface{}{
+										"initContainers": []interface{}{
+											map[string]interface{}{
+												"name":  "download-jdbc-driver",
+												"image": version.LatestInitImage,
+												"command": []interface{}{
+													"sh",
+													"-c",
+													"mkdir -p /amq/init/config/extra-libs && curl -o /amq/init/config/extra-libs/postgresql-42.7.1.jar https://jdbc.postgresql.org/download/postgresql-42.7.1.jar",
+												},
+												"volumeMounts": []interface{}{
+													map[string]interface{}{
+														"name":      "amq-cfg-dir",
+														"mountPath": "/amq/init/config",
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						},
+					},
+				}
+
+				peerB := &brokerv1beta1.ActiveMQArtemis{}
+
+				// a clone of peerA
+				peerA.DeepCopyInto(peerB)
+				peerB.Name = peerPrefix + "-peer-b"
+
+				By("provisioning the broker peer-a")
+				Expect(k8sClient.Create(ctx, &peerA)).Should(Succeed())
+
+				By("provisioning the broker peer-b")
+				Expect(k8sClient.Create(ctx, peerB)).Should(Succeed())
+
+				By("verifying one broker ready, it is a race to lock the db")
+				Eventually(func(g Gomega) {
+
+					createdBrokerCrd := &brokerv1beta1.ActiveMQArtemis{}
+
+					g.Expect(k8sClient.Get(ctx, types.NamespacedName{
+						Name:      peerPrefix + "-peer-a",
+						Namespace: defaultNamespace}, createdBrokerCrd)).Should(Succeed())
+					if verbose {
+						fmt.Printf("\npeer-a CR Status:%v", createdBrokerCrd.Status)
+					}
+					readyPeer := meta.IsStatusConditionTrue(createdBrokerCrd.Status.Conditions, brokerv1beta1.ReadyConditionType)
+
+					if !readyPeer {
+						g.Expect(k8sClient.Get(ctx, types.NamespacedName{
+							Name:      peerPrefix + "-peer-b",
+							Namespace: defaultNamespace}, createdBrokerCrd)).Should(Succeed())
+						if verbose {
+							fmt.Printf("\npeer-b CR Status:%v", createdBrokerCrd.Status)
+						}
+						readyPeer = meta.IsStatusConditionTrue(createdBrokerCrd.Status.Conditions, brokerv1beta1.ReadyConditionType)
+					}
+					g.Expect(readyPeer).Should(BeTrue())
+
+				}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+				brokerService := "broker-ha-jdbc"
+				By("provisioning loadbalanced service for this CR, for use within the cluster via dns")
+				svc := &corev1.Service{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Service",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      brokerService,
+						Namespace: defaultNamespace,
+					},
+					Spec: corev1.ServiceSpec{
```

Comment:
publishNotReadyAddresses: true could speed up an eventual client failover

Reply:
possibly, it may hinder normal operation too, we need to check the impact on the service if there are "not ready" pods in the mix.